internal/db/db.go

```diff
@@ -0,0 +1,235 @@
+// Package db предоставляет функционал для работы с SQLite базой данных.
+//
+// Пакет обеспечивает:
+// - Подключение к SQLite базе данных
+// - Инициализацию схемы базы данных
+// - CRUD операции для задач планировщика
+// - Проверку существования и валидацию данных
+//
+// Основные функции:
+//
+// Подключение к БД:
+//   - Connect(*config.Config) - устанавливает соединение с БД
+//   - CheckAndCreate(string) - создаёт необходимые директории и файл БД
+//
+// Работа с задачами:
+//   - AddTask(date, title, comment, repeat string, *sql.DB) - добавление новой задачи
+//   - GetTasks(*sql.DB, search string, limit int) - получение списка задач
+//   - GetTask(*http.Request, *sql.DB) - получение одной задачи по ID
+//   - UpdateTask(*sql.DB, dto.Task) - обновление существующей задачи
+//   - DeleteTask(*sql.DB, id string) - удаление задачи
+
+// Все операции с базой данных защищены от SQL-инъекций благодаря
+// использованию подготовленных выражений и параметризованных запросов.
+package db
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"go_final_project/internal/config"
+	"go_final_project/internal/dto"
+	"go_final_project/internal/utils"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	_ "modernc.org/sqlite"
+)
+
+// CheckAndCreate проверяет существование директории и файла по указанному пути dbPath.
+// Если они отсутствуют, создаёт их.
+//
+// Выполняет следующие операции:
+// 1. Проверяет существование директории для БД
+// 2. Создаёт директорию с правами 0755, если она отсутствует
+// 3. Проверяет существование файла БД
+// 4. Создаёт пустой файл БД, если он отсутствует
+func CheckAndCreate(dbPath string) error {
+	dir := filepath.Dir(dbPath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		createErr := os.MkdirAll(dir, 0755)
+		if createErr != nil {
+			return fmt.Errorf("Ошибка создания директоррии: %s\n", createErr)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return fmt.Errorf("Ошибка создания директоррии: %w", createErr)
```